src/camera_utils.py

```diff
@@ -108,24 +108,37 @@ def compute_image_footprint_on_surface(
 
 
 def compute_ground_sampling_distance(
-    camera: Camera, distance_from_surface: float
+    camera: Camera, distance_from_surface: float, camera_angle_rad: float = 0.0
 ) -> float:
     """Compute the ground sampling distance (GSD) at a given distance from the surface.
 
     Args:
         camera: the camera model.
         distance_from_surface: distance from the surface (in m).
+        camera_angle_rad: angle of the camera from nadir (in radians). Default is 0 (nadir).
 
     Returns:
-        The GSD in meters (smaller among x and y directions). You should return a float and not a numpy data type.
+        The GSD in meters. For nadir (0°), returns the smaller GSD (higher resolution).
+        For non-nadir angles, returns the GSD in the direction of tilt (Y direction),
+        as this is the limiting factor for motion blur. You should return a float and not a numpy data type.
     """
     
-    # Get the image footprint
-    footprint = compute_image_footprint_on_surface(camera, distance_from_surface)
+    # Get the image footprint (accounting for camera angle if non-zero)
+    if camera_angle_rad == 0.0:
+        footprint = compute_image_footprint_on_surface(camera, distance_from_surface)
+    else:
+        # Import here to avoid circular dependency
+        from src.plan_computation import compute_non_nadir_footprint
+        footprint = compute_non_nadir_footprint(camera, distance_from_surface, camera_angle_rad)
```

Comment:
I guess compute_non_nadir_footprint for angle input = 0 does not return the same value as compute_image_footprint_on_surface because of the min operation we are doing?